File: guilherme/Problema0358/Problema0358.py
print("Digite a quantidade de clientes do Banco A: ")
a = int(input())
x = list(range(a))
for i in range(a):
    print("Digite o nome do ",i,"º cliente: ")
    x[i] = input()

for i in range(a):
    for j in range(a):
        if(x[i] < x[j]):
            apoio = x[i]
            x[i] = x[j]
            x[j] = apoio
print("\nVetor X: ")
for i in range(a):
    print(x[i])

print("Digite a quantidade de clientes do Banco B: ")
b = int(input())
y = list(range(b))
for i in range(b):
    print("Digite o nome do ",i,"º cliente: ")
    y[i] = input()

for i in range(b):
    for j in range(b):
        if(y[i] < y[j]):
            apoio = y[i]
            y[i] = y[j]
            y[j] = apoio
print("\nVetor Y: ")
for i in range(b):
    print(y[i])

z = list(range(a+b))
for i in range(a):
    z[i] = x[i]
temp = 0
for i in range(a, a+b):
    z[i] = y[temp]
    temp += 1
for i in range(a+b):
    for j in range(i+1, a+b):
        if(z[i] > (z[j])):
            apoio = z[i]
            z[i] = z[j]
            z[j] = apoio
        
        if(z[i] == (z[j])):
            if(j+1 < a+b):
                apoio = z[j]
                z[j] = z[j+1]
                z[j+1] = apoio
            else:
                z[j] = "-"
            
print("\nVetor Z: ")
for i in range(a+b):
    if(z[i] != "-"):
        print(z[i])

File: guilherme/Problema0358/Problema0358.java
import java.util.Scanner;

public class Problema0358{
    static Scanner input = new Scanner(System.in);
    public static void main(String[] args) {
        String[] x, y, z;
        String apoio;
        int a, b;
        System.out.println("Digite a quantidade de clientes do Banco A: ");
        a = Integer.valueOf(input.nextLine());
        x = new String[a];
        for(int i = 0; i < a; i++){
            System.out.println("Digite o nome do " + i + "º cliente: ");
            x[i] = input.nextLine();
        }
        for(int i = 0; i < a; i++){
            for(int j = i + 1; j < a; j++){
                if(x[i].compareTo(x[j]) > 0){
                    apoio = x[i];
                    x[i] = x[j];
                    x[j] = apoio;
                }
            }
        }
        System.out.println("Lista X:");
        for(int i = 0; i < a; i++){
            System.out.println(x[i]);
        }


        System.out.println("Digite a quantidade de clientes do Banco B: ");
        b = Integer.valueOf(input.nextLine());
        y = new String[b];
        for(int i = 0; i < b; i++){
            System.out.println("Digite o nome do " + i + "º cliente: ");
            y[i] = input.nextLine();
        }
        for(int i = 0; i < b; i++){
            for(int j = i + 1; j < b; j++){
                if(y[i].compareTo(y[j]) > 0){
                    apoio = y[i];
                    y[i] = y[j];
                    y[j] = apoio;
                }
            }
        }
        System.out.println("Lista Y:");
        for(int i = 0; i < b; i++){
            System.out.println(y[i]);
        }

        z = new String[a+b];
        for(int i = 0; i < a; i++){
            z[i] = x[i];
        }
        int temp = 0;
        for(int i = a; i < a+b; i++){
            z[i] = y[temp];
            temp++;
        }
        for(int i = 0; i < a+b; i++){
            for(int j = i + 1; j < a+b; j++){
                if(z[i].compareTo(z[j]) > 0){
                    apoio = z[i];
                    z[i] = z[j];
                    z[j] = apoio;
                }
                if(z[i].compareTo(z[j]) == 0){
                    if(j+1 < a+b){
                        apoio = z[j];
                        z[j] = z[j+1];
                        z[j+1] = apoio;
                    }else{
                        z[j] = "-";
                    }
                }
            }
        }
        System.out.println("Lista Z:");
        for(int i = 0; i < a+b; i++){
            if(z[i].compareTo("-") != 0)
                System.out.println(z[i]);
        }
    }
}